### ISSUE TITLE
Meshing is sometimes not continuous across chunk boundaries when it should be

### ISSUE THREAD

**Issue by ceesem** [seung-lab/PyChunkedGraph#239]
In some cases, the meshing does not create continuous meshes across chunk boundaries. What seems to happen is that one side of the mesh is left open (correctly) while the other is closed as if the object ends, thus it seems like an issue with how cross-chunk edges are being handled.

Here is one example: https://akhilesh-graphene-sharded-dot-neuromancer-seung-import.appspot.com/?json_url=https://globalv1.daf-apis.com/nglstate/api/v1/4917032483028992

This is a **critical issue**, since disconnected meshes make it impossible for us to properly analyze neuronal morphology without ad hoc fixes that are specific due to what should be a temporary bug. While the overall frequency of events is low, a single neuron typically has multiple such cases. They seem more likely in thin axon than thicker processes.

**Comment by manuel-castro** [comment]
Mind sharing another example? I accidentally fixed this one.

**Comment by ceesem** [comment]
Here's another one from the same cell: https://akhilesh-graphene-sharded-dot-neuromancer-seung-import.appspot.com/?json_url=https://globalv1.daf-apis.com/nglstate/api/v1/5770102981001216

**Comment by manuel-castro** [comment]
> Here's another one from the same cell: https://akhilesh-graphene-sharded-dot-neuromancer-seung-import.appspot.com/?json_url=https://globalv1.daf-apis.com/nglstate/api/v1/5770102981001216
> 
> These were the two I had handy. I can probably come up with a good workflow for finding them if you need more examples.

Knowing how to find more examples would be great. Do you think you could explain how one might do that?

**Comment by ceesem** [comment]
My workflow starts with downloading the mesh and cleaning it up a bit and add "merge edges" which span proofreading events whether the segmentation is connected or not. Then I skeletonize it, run connected components on the skeleton, and look for skeleton tips that are not in the same connected component.

(note that this code uses `annotationframeworkclient` branch `updated_imagery` and `meshparty` branch `meshwork_merge`)

```python
from meshparty import trimesh_io, mesh_filters, skeletonize
from annotationframeworkclient import FrameworkClient

datastack_name = 'minnie65_phase3_v1'
client = FrameworkClient(datastack_name=datastack_name)

# Out of a suspicion that this affects proofread cells more, let's stick to some concrete examples
proofread_ids = [864691134884807418, 864691135256248239, 864691135272206865,
       864691135341199281, 864691135367033849, 864691135385422677,
       864691135446675028, 864691135462811710, 864691135517422218,
       864691135583884664, 864691135718541617, 864691135724333739,
       864691135866576150, 864691135988348672, 864691136084134252,
       864691136108768952, 864691136116205476, 864691136118489624,
       864691136227020113, 864691136378815445, 864691136521633681,
       864691136903144370, 864691136951664863, 864691137019596142]

# Download one of the meshes
oid = proofread_ids[0]
mm = trimesh_io.MeshMeta(cv_path='graphene://https://minniev1.microns-daf.com/segmentation/table/minnie3_v1')
mesh = mm.mesh(seg_id=oid)

# Add link edges which span merge events
mesh.add_link_edges(seg_id=oid, client=client.chunkedgraph)
lcc = mesh_filters.filter_components_by_size(mesh, min_size=200)
meshf = mesh.apply_mask(lcc)

# Skeletonize and look at connected components
from scipy import sparse, spatial
sk = skeletonize.skeletonize_mesh(meshf)
ccs, lbls = sparse.csgraph.connected_components(sk.csgraph)

eps = sk.end_points_undirected
ep_labels = lbls[sk.end_points_undirected]

d_thresh = 1000
all_jump_locs = []

for lbl in np.unique(lbls):
    within_ends = eps[ep_labels == lbl]
    other_ends = eps[ep_labels > lbl]
    
    if len(within_ends) > 0 and len(other_ends) > 0:
        kdtree = spatial.cKDTree(sk.vertices[within_ends])
        d, inds = kdtree.query(sk.vertices[other_ends])

        jump_locs = sk.vertices[other_ends[d < d_thresh]]
        if len(jump_locs)>0:
            all_jump_locs.append(jump_locs)
all_jump_locs = np.concatenate(all_jump_locs)
```
Note that the locations at the end here would be in nm coordinates, not voxels.

For looking this case, for cell `864691134884807418`, I get the voxel locations:
```
array([[300625., 147766.,  23041.],
       [319593.,  97471.,  22842.],
       [322591., 161605.,  19667.],
       [331296.,  99036.,  20048.],
       [297890., 132184.,  22035.],
       [302625., 146611.,  22818.],
       [319000.,  94153.,  22885.],
       [333343.,  99461.,  18485.]])
```

The ones I visually checked all looked like this type of error.

**Comment by ceesem** [comment]
Any luck?

**Comment by manuel-castro** [comment]
None so far. I haven't been able to reproduce the error locally yet. When I run the remeshing code locally it seems to always produce the correct mesh. Frustrating but I'll keep looking into it.

**Comment by ceesem** [comment]
Hm, that is always frustrating. I forget, is there a call to do remeshing on a specific chunk or root id in the deployed Minnie PyChunkedGraph? And if so, does it have this problem consistently?

**Comment by manuel-castro** [comment]
I hadn't tried pinging the remeshing endpoint directly yet: there is an endpoint that takes a list of L2 ids to remesh. That's what I'm figuring out how to do now

**Comment by ceesem** [comment]
We just added (for the Minnie deployment at least) an endpoint to get the level-2 chunk graph for a root id, if that helps. You can get it at `/segmentation/api/v1/table/{table_name}/node/{root_id}/lvl2_graph`.

**Comment by manuel-castro** [comment]
> Hm, that is always frustrating. I forget, is there a call to do remeshing on a specific chunk or root id in the deployed Minnie PyChunkedGraph? And if so, does it have this problem consistently?

Triggering the remeshing job on the deployed ChunkedGraph also produces the correct mesh. I have found one timestamp issue that could possibly be related to this problem, so I will fix that today. If we're lucky that will solve this issue. I've noticed that some "initial" meshes also have this problem, so I'll do some debugging later with the initial meshing code as well (which is in igneous).
